src/vcf/assembly_checker.cpp

```diff
@@ -67,19 +77,31 @@ namespace ebi
               }
 
               RecordCore record_core = build_record_core(line,line_num);
+              auto synonyms_list = synonyms_map.get_synonym_list(record_core.chromosome);
 
-              if (index.count(record_core.chromosome) == 0) {
-                  report_missing_chromosome(line_num,record_core,outputs);
+              if (record_core.position == 0) {
+                  report_telomere_position(line_num,outputs);
                   continue;
               }
 
-              if (record_core.position == 0) {
-                  report_telomere_position(line_num,outputs);
+              std::string found_synonym = "no_match";
+              for (auto chrom_synonym : synonyms_list.list) {
+                  if (index.count(chrom_synonym) != 0) {
+                      if (found_synonym != "no_match") {
```

Comment:
@jmmut in case we are to extract this loop out. we will have to return two things, found_synonym and a bool value. could it be ok if returning "" could tell us that there is no match ?

Reply:
sometimes there are cells with "na" which means "not available" so the empty string could be confused between "not found" and "found but not available".

you can return by reference, or return a tuple

Reply:
fine with returning touple. but it seems bit bulky in reading functions, approach. passing a reference of bool seems better approach. A fuction was designed to give a single output. getting multiple outputs destroy the beauty of the function. still I am ok with that provided there is no other better way.